src/server/authentication.rs

```diff
@@ -0,0 +1,36 @@
+/*
```

Comment:
I'd like to keep the server methods and service name the same -- the corresponding functions here come from `password/` so I think that would be a better name for this submodule.